### ISSUE TITLE
For loop with braces inside command substitution breaks syntax highlighting

### ISSUE THREAD

**Issue by yusufaktepe** [jeff-hykin/better-shell-syntax#31]
Here is an example snippet:
```bash
A=$(
    for ((i=0; i<${#ID[@]}; i++)); {
       do_something
    }
)
```
These are fine:
```bash
A=$(
    for ((i=0; i<${#ID[@]}; i++)); do
        do_something
    done
)
```

```bash
A=$(\
    for ((i=0; i<${#ID[@]}; i++)); {
        do_something
    }
)
```

**Comment by jeff-hykin** [comment]
This has to do with a legacy version of the `for` pattern that I haven't converted yet (someone else originally wrote it) so it might take me a bit to get to.

Sadly your last example is actually a problem of its own (`for` is recognized as a command instead of a keyword) so I'll add that to this issue

**Comment by jeff-hykin** [comment]
fixed! v1.7.0
